query_execution/WorkOrdersContainer.hpp

```diff
@@ -220,8 +226,9 @@ class WorkOrdersContainer {
    * @param operator_index The index of the operator in the query DAG.
    **/
   void addNormalWorkOrder(WorkOrder *workorder, const std::size_t operator_index) {
-    DEBUG_ASSERT(workorder != nullptr);
-    DEBUG_ASSERT(operator_index < num_operators_);
+    DCHECK(workorder != nullptr);
+    DCHECK(operator_index < num_operators_);
+    workorder->setQueryID(query_id_);
```

Comment:
That's a good point. RebuildWorkOrder has query_id in its constructor because in the execute method, it needs the ``query_id`` for sending a pipelining message, which in turn needs the query ID. 

Adding the query ID to all WorkOrder constructors requires modifying base relational operator, base WorkOrder and all the constructor calls to the derived WorkOrder classes. I can mark this task as a TODO and return to it once this PR gets merged. Let me know if that sounds alright.